web/packages/teleterm/src/ui/DocumentGatewayApp/AppGateway.tsx

```diff
@@ -69,38 +95,54 @@ export function AppGateway(props: {
         </ButtonSecondary>
       </Flex>
 
-      {props.disconnectAttempt.status === 'error' && (
-        <Alert details={props.disconnectAttempt.statusText}>
+      {disconnectAttempt.status === 'error' && (
+        <Alert details={disconnectAttempt.statusText}>
           Could not close the connection
         </Alert>
       )}
 
-      <Flex as="form" ref={formRef} gap={2}>
+      <Flex as="form" gap={2}>
         <Validation>
           <PortFieldInput
-            label="Port"
+            label={
+              <LabelWithAttemptStatus
+                text="Local Port"
+                attempt={changeLocalPortAttempt}
+              />
+            }
             defaultValue={gateway.localPort}
-            onChange={e => handleChangePort(e.target.value)}
+            onChange={handleLocalPortChange}
             mb={2}
           />
+          {isMultiPort && (
+            <PortFieldInput
+              label={
+                <LabelWithAttemptStatus
+                  text="Target Port"
+                  attempt={changeTargetPortAttempt}
+                />
+              }
+              required
+              defaultValue={gateway.targetSubresourceName}
+              onChange={handleTargetPortChange}
+              mb={2}
+            />
+          )}
         </Validation>
-        {props.changePortAttempt.status === 'processing' && (
-          <Indicator
-            size="large"
-            pt={3} // aligns the spinner to be at the center of the port input
-            css={`
-              display: flex;
-            `}
-          />
-        )}
       </Flex>
 
       <Text>Access the app at:</Text>
       <TextSelectCopy my={1} text={address} bash={false} />
 
-      {props.changePortAttempt.status === 'error' && (
-        <Alert details={props.changePortAttempt.statusText}>
-          Could not change the port number
+      {changeLocalPortAttempt.status === 'error' && (
+        <Alert details={changeLocalPortAttempt.statusText}>
```

Comment:
Another nit :) I think we could improve the vertical spacing between elements (but only if it is not too much work):
| Now  | With changes |
|---|---|
|<img width="770" alt="image" src="https://github.com/user-attachments/assets/83279c25-bf4c-4bc1-b80b-a2378fe23cae" /> |  <img width="770" alt="image" src="https://github.com/user-attachments/assets/bd9e70bd-bcfa-4064-8612-633fc9bd8b4b" /> |


BTW, it looks like we don't handle the expired certs error for `changeTargetPort`, could you wrap it in `retryWithRelogin`?